### ISSUE TITLE
Invalid file references in tsconfig.json lead to misleading ts-loader behaviour

### ISSUE THREAD

**Issue by johnnyreilly** [TypeStrong/ts-loader#117]
Hey,

Having just burned a good amount of time on something that turned out to be nothing I thought I'd share my learnings.

Consider the following tsconfig.json:

```
{
    "compileOnSave": false,
    "filesGlob": [
        "**/*.{ts,tsx}",
        "../typings/**/*.*.ts"
    ],
    "compilerOptions": {
        "target": "es6",
        "module": "commonjs",
        "noImplicitAny": false,
        "suppressImplicitAnyIndexErrors": true,
        "removeComments": false,
        "preserveConstEnums": true,
        "sourceMap": true
    },
    "files": [
        "i-am-a-file-what-does-not-exist.ts",
        "demo/components/Module.tests.tsx",
        "lib/index.tests.ts",
        "../typings/flux/flux.d.ts",
        "../typings/jasmine/jasmine.d.ts",
        "../typings/node/node.d.ts",
        "../typings/react-router/history.d.ts",
        "../typings/react-router/react-router.d.ts",
        "../typings/react/react-addons-test-utils.d.ts",
        "../typings/react/react-dom.d.ts",
        "../typings/react/react.d.ts",
        "../typings/tsd.d.ts"
    ],
    "exclude": [],
    "atom": {
        "rewriteTsconfig": true
    }
}
```

Spot the invalid file reference?  I didn't either.  For hours.  This is for my tests and Karma hooks up webpack / TS loader.  The following errors were then thrown:

```
11 12 2015 06:14:50.221:ERROR [karma]: { [Error: no such file or directory]
  code: 'ENOENT',
  errno: 34,
  message: 'no such file or directory',
  path: '/_karma_webpack_/test/lib/index.tests.ts' }
Error: no such file or directory
    at MemoryFileSystem.readFileSync (C:\Source\globalize-so-what-cha-want\node_modules\memory-fs\lib\MemoryFileSystem.j
s:107:10)
    at MemoryFileSystem.readFile (C:\Source\globalize-so-what-cha-want\node_modules\memory-fs\lib\MemoryFileSystem.js:29
7:21)
    at doRead (C:\Source\globalize-so-what-cha-want\node_modules\karma-webpack\index.js:156:26)
    at Plugin.readFile (C:\Source\globalize-so-what-cha-want\node_modules\karma-webpack\index.js:160:3)
    at doNTCallback0 (node.js:419:9)
    at process._tickCallback (node.js:348:13)
11 12 2015 06:14:50.221:ERROR [karma]: { [Error: no such file or directory]
  code: 'ENOENT',
  errno: 34,
  message: 'no such file or directory',
  path: '/_karma_webpack_/test/demo/components/Module.tests.tsx' }
Error: no such file or directory
    at MemoryFileSystem.readFileSync (C:\Source\globalize-so-what-cha-want\node_modules\memory-fs\lib\MemoryFileSystem.j
s:107:10)
    at MemoryFileSystem.readFile (C:\Source\globalize-so-what-cha-want\node_modules\memory-fs\lib\MemoryFileSystem.js:29
7:21)
    at doRead (C:\Source\globalize-so-what-cha-want\node_modules\karma-webpack\index.js:156:26)
    at Plugin.readFile (C:\Source\globalize-so-what-cha-want\node_modules\karma-webpack\index.js:160:3)
    at doNTCallback0 (node.js:419:9)
    at Immediate._tickCallback [as _onImmediate] (node.js:348:13)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
```

This didn't guide me back (unfortunately) to the actual problem.  Both the files mentioned above **do** exist.   I wonder if it might be worth getting ts-loader to shout loud specifically about non existent files *if* tsconfig.json contains a reference to a non-existent file given the painful side-effects?

**Comment by jbrantly** [comment]
/agree

Apologies for the wasted time :frowning:

**Comment by johnnyreilly** [comment]
No worries chap :+1:

**Comment by johnnyreilly** [comment]
Actually, would you mind if I had a crack at this?  Can't do it right now but I want to try and give back in some way..

I haven't looked at the source a great deal but it's probably fairly simple to implement this?

**Comment by jbrantly** [comment]
Sounds good! Yea, probably not too bad to implement this. There are a couple of examples in the code of adding errors that result from tsconfig.json.

**Comment by episodeyang** [comment]
just ran into the same problem and this post saved me some time.

I'm switching from browserify to web pack. It looks like the browserify ts loader automatically ignores the files list in the tsconfig.json so this was not a problem for me before switching to web pack.

What is the get around atm?

**Comment by jbrantly** [comment]
@episodeyang I'm not sure I understand your question. If you're asking how you can prevent the error above, it would be to remove the invalid file reference in tsconfig.json. The issue here is that the current error reporting doesn't actually tell you *which* file is invalid, so you have to figure that out yourself.

**Comment by jbrantly** [comment]
Published in v0.8.1. Thanks again @johnnyreilly!

**Comment by johnnyreilly** [comment]
My pleasure!
